machine/os/ubuntu.go

```diff
@@ -0,0 +1,37 @@
+package os
+
+import (
+	"strings"
+
+	"github.com/docker/machine/drivers"
+)
+
+func init() {
+	RegisterRuntime("ubuntu", &RegisteredRuntime{
+		Detect: UbuntuDetection,
```

Comment:
As noted in IRC, I'd prefer the actual function that does the detection (`CompatibleWithHost(h)`?) to be part of a generic "Provisioner" interface.

This should totally accept a factory function like the host drivers do, though (`New: NewUbuntuProvisioner` style).